File: LC0199_Binary_Tree_Right_Side_View/binary_tree_right_side_view.go
package main

import (
	"container/list"
)

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func rightSideView(root *TreeNode) []int {
	// trivial case
	if root == nil {
		return []int{}
	}

	// Perform level-order traversal on the tree, and the last node in the current level is the only visible node of the level.
	answer := make([]int, 0)
	queue := list.New()
	queue.PushBack(root)
	n := queue.Len()
	for n > 0 {
		answer = append(answer, queue.Back().Value.(*TreeNode).Val)
		for i := 0; i < n; i++ {
			head := queue.Front()
			curr_node := head.Value.(*TreeNode)
			queue.Remove(head)
			if curr_node.Left != nil {
				queue.PushBack(curr_node.Left)
			}
			if curr_node.Right != nil {
				queue.PushBack(curr_node.Right)
			}
		}
		n = queue.Len()
	}
	return answer
}

File: LC0199_Binary_Tree_Right_Side_View/binary_tree_right_side_view.py
from typing import Optional, List
from collections import deque

# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        # trivial case
        if not root:
            return []
            
        # Perform level-order traversal on the tree, and the last node we see in each level is the only visible node of that level.
        answer = []
        queue = deque()
        queue.append(root)
        while queue:
            answer.append(queue[-1].val)
            n = len(queue)
            for i in range(n):
                curr_node = queue.popleft()
                if curr_node.left:
                    queue.append(curr_node.left)
                if curr_node.right:
                    queue.append(curr_node.right)
        return answer